ext/imap/php_imap.stub.php

```diff
@@ -0,0 +1,354 @@
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_sort($stream_id, int $criteria, int $reverse, int $options = 0, string $search_criteria ='', string $charset = '') {}
+
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_uid($stream_id, int $msg_no) {}
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_msgno($stream_id, int $unique_msg_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_list($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_lsub($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_getsubscribed($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_getmailboxes($stream_id, string $ref, string $pattern) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_fetch_overview($stream_id, string $sequence, int $options = 0) {}
+
+function imap_alerts(): array {}
+
+function imap_errors(): array {}
+
+/**
+ * @return string|false
+ */
+function imap_last_error() {}
+
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_search($stream_id, string $criteria, int $options = 0, string $charset = '') {}
+
+/**
+ * @return string|false
+ */
+function imap_utf7_decode(string $buf) {}
+
+function imap_utf7_encode(string $buf): string {}
+
+#ifdef HAVE_IMAP_MUTF7
+/**
+ * @return string|false
+ */
+function imap_utf8_to_mutf7(string $in) {}
+
+/**
+ * @return string|false
+ */
+function imap_mutf7_to_utf8(string $in) {}
+#endif
+
+/**
+ * @return array|false
+ */
+function imap_mime_header_decode(string $str) {}
+
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_thread($stream_id, int $options = 0) {}
+
+/**
+ * @return int|bool
+ */
+function imap_timeout(int $timeout_type, int $timeout = -1) {}
+
+#if defined(HAVE_IMAP2000) || defined(HAVE_IMAP2001)
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_get_quota($stream_id, string $qroot) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_get_quotaroot($stream_id, string $mbox) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_set_quota($stream_id, string $qroot, int $mailbox_size): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_setacl($stream_id, string $mailbox, string $id, string $rights): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_getacl($stream_id, string $mailbox) {}
+#endif
+
+function imap_mail(string $to, string $subject, string $message, string $additional_headers = '', string $cc = '', string $bcc = '', string $rpath = ''): bool {}
```

Comment:
It seems all optional params default to `UNKNOWN` (either they're given or not):
```suggestion
function imap_mail(string $to, string $subject, string $message, string $additional_headers = UNKNOWN, string $cc = UNKNOWN, string $bcc = UNKNOWN, string $rpath = UNKNOWN): bool {}
```